tools/gmhazard_scripts/db_creation/empirical_db/common.py

```diff
@@ -170,6 +172,8 @@ def write_data_and_close(
             ims=np.asarray(im_list, dtype=str),
         )
         imdb_dict[imdb_keys].close()
+        if rupture_lookup:
+            imdb_dict[imdb_keys].add_rupture_lookup(imdb_dict[imdb_keys].db_ffp, 1)
```

Comment:
Don't think this should be after the close?

This is potentially dangerous as it shouldn't run for ds (it will quickly run out of RAM I think - if not take forever to complete). We should add some checking to this method to only do it if it is a fault db and error otherwise.

Reply:
Oh, the method does open it? It's not clear from a cursory glance. If it's a larger db it probably needs more than 1 processor too. But fault normally only runs in a single process though.

Reply:
@joelridden Agree with @jasonmotha on this one, probably good to add assert to ensure this only runs for fault IMDB generation

Reply:
@claudio525 Not sure if an assert is necessary as the rupture lookup variable is default to false and can currently only be set from the calc_emp_flt script, therefore it won't actually run for DS unless you manually grab a DS db and run that function.
At that point though I feel like its the users fault if they are trying to run this function for DS? Maybe a comment about it saying a warning for running this on any DS db?

Reply:
Ah yeah, that sounds all good